### PR TITLE
atom: add missing methods

### PR DESCRIPTION
**Pull request by idleberg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/atom/index.d.ts [hunk 1]
```diff
@@ -199,6 +202,18 @@ export interface AtomEnvironment {
...
+     * Restores the full screen and maximized state after the window has resized to prevent resize
+     * glitches.
+     */
+    displayWindow(): void;
+
+    /** Get the dimensions of this window. */
+    getWindowDimensions(): { x: number, y: number, width: number, height: number };
+
+    /** Set the dimensions of the window. */
+    setWindowDimensions(dimensions: object): Promise<object>;
```
Comment (GlenCFL) [inline types/atom/index.d.ts]: Going by the [source code](https://github.com/atom/atom/blob/v1.36.0/src/atom-environment.js#L704), we can define the type of this parameter as: `{ x?: number, y?: number, height?: number, width?: number }`.

File: types/atom/index.d.ts [hunk 2]
```diff
@@ -121,6 +121,9 @@ export interface AtomEnvironment {
     /** Returns a boolean that is true if the current window is running specs. */
     inSpecMode(): boolean;
 
+    /** Get the full name of this Atom release (e.g. "Atom", "Atom Beta") */
+    getAppName(): string;
```
Comment (GlenCFL) [inline types/atom/index.d.ts]: This function was first available in Atom version `v1.40.0`. To bump the definition version from `v1.36.0` to `v1.40.0`, you'll need to replace the three occurrences of `1.36` within the file with `1.40`.

**Review by GlenCFL — CHANGES_REQUESTED** [review]
Hi, firstly thank you for your second pull request helping to maintain this definition file. I'm going to have to request a few changes with this one though, though each should be quick.

**Comment by idleberg** [comment]
@GlencCFL Thanks for the guidance. I hope I can make more regular and more systematic contributions in the future. This PR was mostly triggered by personal need.

**Comment by GlenCFL** [comment]
@idleberg You seem to have misunderstood this one:
```ts
getWindowDimensions(): { x: number, y: number, width: number, height: number };

setWindowDimensions(dimensions: { x?: number, y?: number, width?: number, height?: number }): Promise<object>;
```

**Review by GlenCFL — APPROVED** [review]
Looks good. Thanks again.

**Pull request merged**